docs(project): adding project specific files

mynktl opened the pull request:
Changes:
- This PR adds project specific files to guide contributors
    List of the files is as below: 
     - [.github/ISSUE_TEMPLATE/bug_report.md](https://github.com/openebs/cstor/blob/f6c60891c02d7e38bd21343f151ccfbbd2d9e0df/.github/ISSUE_TEMPLATE/bug_report.md)
     - [.github/ISSUE_TEMPLATE/feature-request.md](https://github.com/openebs/cstor/blob/f6c60891c02d7e38bd21343f151ccfbbd2d9e0df/.github/ISSUE_TEMPLATE/feature-request.md)
     - [.github/pull_request_template.md](https://github.com/openebs/cstor/blob/f6c60891c02d7e38bd21343f151ccfbbd2d9e0df/.github/pull_request_template.md)
     - [CODE_OF_CONDUCT.md](https://github.com/openebs/cstor/blob/f6c60891c02d7e38bd21343f151ccfbbd2d9e0df/CODE_OF_CONDUCT.md)
     - [CONTRIBUTING.md](https://github.com/openebs/cstor/blob/f6c60891c02d7e38bd21343f151ccfbbd2d9e0df/CONTRIBUTING.md)
     - [GOVERNANCE.md](https://github.com/openebs/cstor/blob/f6c60891c02d7e38bd21343f151ccfbbd2d9e0df/GOVERNANCE.md)
     - [RELEASE.md](https://github.com/openebs/cstor/blob/f6c60891c02d7e38bd21343f151ccfbbd2d9e0df/RELEASE.md)
     - [code-standard.md](https://github.com/openebs/cstor/blob/f6c60891c02d7e38bd21343f151ccfbbd2d9e0df/code-standard.md)
     - [developer-setup.md](https://github.com/openebs/cstor/blob/f6c60891c02d7e38bd21343f151ccfbbd2d9e0df/developer-setup.md)

Following files were removed due to duplication:
     - .github/PULL_REQUEST_TEMPLATE.md
     - .github/ISSUE_TEMPLATE.md

Signed-off-by: mayank <mayank.patel@mayadata.io>

<!--- Provide a general summary of your changes in the Title above -->
<!--- Explain how the fix was tested -->

Commit:
Adding following files for the project:
- .github/ISSUE_TEMPLATE/bug_report.md
- .github/ISSUE_TEMPLATE/feature-request.md
- .github/pull_request_template.md
- CODE_OF_CONDUCT.md
- CONTRIBUTING.md
- GOVERNANCE.md
- RELEASE.md
- code-standard.md
- developer-setup.md

Following files removed due to duplication:
- .github/PULL_REQUEST_TEMPLATE.md
- .github/ISSUE_TEMPLATE.md

Signed-off-by: mayank <mayank.patel@mayadata.io>

vishnuitta commented on .github/ISSUE_TEMPLATE/bug_report.md:
```diff
@@ -0,0 +1,31 @@
+(Pasting long output into a [GitHub gist](https://gist.github.com) or other [Pastebin](https://pastebin.com/) is fine.)
+
+* `kubectl logs deployment/maya-apiserver -n openebs`
+* `kubectl logs deployment/<pv-name>-target -n <openebs_namespace> -c cstor-istgt`
```
does this works with 'deployment'?

mittachaitu replied:
It will work

vishnuitta reviewed (dismissed):
changes are good

mittachaitu commented on .github/ISSUE_TEMPLATE/bug_report.md:
```diff
@@ -0,0 +1,31 @@
+
+* `kubectl logs deployment/maya-apiserver -n openebs`
+* `kubectl logs deployment/<pv-name>-target -n <openebs_namespace> -c cstor-istgt`
+* `kubectl get pods -n openebs`
```
IMO good to get the output of pool configurations
`kubectl get csp -l openebs.io/storage-pool-claim=<spc_name> -o yaml`

mittachaitu commented on README.markdown:
```diff
@@ -2,16 +2,17 @@
 [![codecov](https://codecov.io/gh/zfsonlinux/zfs/branch/master/graph/badge.svg)](https://codecov.io/gh/zfsonlinux/zfs)
```
Can we rename this file to README.md as we are following in all other places?

mynktl replied:
we can rename it, but i don't think it is required. Extension of the file can be different. If you want we can rename it in a different PR.

Commit:
addressing review comments

Signed-off-by: mayank <mayank.patel@mayadata.io>

mittachaitu reviewed (dismissed):
Changes are good

Commit:
addressing review comments

Signed-off-by: mayank <mayank.patel@mayadata.io>

mittachaitu commented on .github/ISSUE_TEMPLATE/bug_report.md:
```diff
@@ -0,0 +1,32 @@
+* `kubectl logs deployment/maya-apiserver -n openebs`
+* `kubectl logs deployment/<pv-name>-target -n <openebs_namespace> -c cstor-istgt`
+* `kubectl get pods -n openebs`
+* `kubectl get cvr,spc,csp,cstorvolume -A -o yaml`
```
For reporting issue, I think we need logs of cstor-pool container also

mynktl replied:
yes, i've mentioned in above comment.

mittachaitu replied:
that is target pod istgt container right?

mynktl replied:
https://github.com/openebs/cstor/pull/301/files/a7d4c5c80c518e44fbb2cfcf890a5bfafa8dfb40#diff-74c2649c999338ed7a2833d79d4b25c3R15

mittachaitu replied:
Ok!! I am searching for command...

kmova commented on RELEASE.md:
```diff
@@ -0,0 +1,38 @@
+# Release Process
+cstor follows a monthly release cadence. The scope of the release is determined by contributor availability. The scope is published in the [Release Tracker Projects](https://github.com/openebs/openebs/projects).
```
Link should be: https://github.com/orgs/openebs/projects

mynktl replied:
done

mynktl replied:
for other repo i've drafted PR:
https://github.com/openebs/libcstor/pull/53
https://github.com/openebs/velero-plugin/pull/71
We will push further doc related fixes to above pr and will proceed with merge during RC1.

Commit:
fixing project tracker link for openebs

Signed-off-by: mayank <mayank.patel@mayadata.io>

mittachaitu commented on RELEASE.md:
```diff
@@ -0,0 +1,38 @@
+Images are published at the following location:
+https://quay.io/repository/openebs/cstor?tab=tags
+https://hub.docker.com/r/openebs/cstor/tags
+
```
We need to add arm builds also `https://hub.docker.com/r/openebs/cstor-pool-arm64/tags`(I missed in istgt I will add it)

mynktl replied:
done

Commit:
- Fixing links in CONTRIBUTING.md
- updating code-standard.md with changelog example
- updating RELEASE.md about changelog instruction and image location
- updating .github/pull_request_template.md with checklist

Signed-off-by: mayank <mayank.patel@mayadata.io>
